src/components/network-area-diagram-viewer/diagram-utils.ts

```diff
@@ -622,3 +626,61 @@ function getElementType(element: SVGElement | null): ElementType {
     }
     return ElementType.UNKNOWN;
 }
+
+// get view box computed starting from node and text positions
+// defined in diagram metadata
+export function getViewBox(
+    nodes: NodeMetadata[] | undefined,
+    textNodes: TextNodeMetadata[] | undefined,
+    svgParameters: SvgParameters
+): ViewBox {
+    const size = { minX: Number.MAX_VALUE, maxX: -Number.MAX_VALUE, minY: Number.MAX_VALUE, maxY: -Number.MAX_VALUE };
+    const nodesMap: Map<string, NodeMetadata> = new Map<string, NodeMetadata>();
+    nodes?.forEach((node) => {
+        nodesMap.set(node.equipmentId, node);
+        size.minX = Math.min(size.minX, node.x);
+        size.maxX = Math.max(size.maxX, node.x);
+        size.minY = Math.min(size.minY, node.y);
+        size.maxY = Math.max(size.maxY, node.y);
+    });
+    textNodes?.forEach((textNode) => {
+        const node = nodesMap.get(textNode.equipmentId);
+        if (node !== undefined) {
+            size.minX = Math.min(size.minX, node.x + textNode.shiftX);
+            size.maxX = Math.max(size.maxX, node.x + textNode.shiftX + svgParameters.getTextBoxWidth());
```

Comment:
You can put the default constant directly in diagram-utils.ts, it is not a parameter of SvgParameter
```suggestion
            size.maxX = Math.max(size.maxX, node.x + textNode.shiftX + TEXT_BOX_WIDTH_DEFAULT);
```

Reply:
parameter moved to diagram-utils